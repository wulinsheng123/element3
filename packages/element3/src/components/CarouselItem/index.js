import ElCarouselItem from './src/CarouselItem.vue'
/* istanbul ignore next */
ElCarouselItem.install = function (app) {
  app.component(ElCarouselItem.name, ElCarouselItem)
}

export { ElCarouselItem }
