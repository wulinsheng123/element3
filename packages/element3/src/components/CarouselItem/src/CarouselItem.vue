<template>
  <div
    class="el-carousel__item"
    :style="itemStyle"
    :class="{
      'is-active': active,
      'is-animating': animating
    }"
  >
    <div class="el-carousel__mask"></div>
    <slot></slot>
  </div>
</template>

<script>
import { props } from './props.ts'
import { handleChildMethods } from './use'
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'ElCarouselItem',
  props,
  setup() {
    const { translateItem, active, animating, itemStyle } = handleChildMethods()
    return {
      translateItem,
      active,
      animating,
      itemStyle
    }
  }
})
</script>
