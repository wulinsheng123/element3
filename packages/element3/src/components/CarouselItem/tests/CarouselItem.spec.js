import CarouselItem from '../src/CarouselItem.vue'
import { mount } from '@vue/test-utils'
import { nextTick, ref } from 'vue'

describe('Carousel.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(CarouselItem)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render when call translateItem', async () => {
    const items = ref([])
    const getChilrenItems = jest.fn((child) => {
      items.value.push(child.proxy)
    })
    mount(CarouselItem, {
      global: {
        provide: {
          CAROUSEL: { getChilrenItems }
        }
      }
    })
    expect(getChilrenItems).toHaveBeenCalled()
    expect(items.value).toHaveLength(1)
  })
  it('test components translateItem of function', () => {
    const wrapper = mount(CarouselItem, {
      global: {
        provide: {
          CAROUSEL: {
            getChilrenItems() {},
            $parent: {
              proxy: {
                type: null,
                activeIndex: 2,
                direction: 'vertical'
              }
            }
          }
        }
      }
    })
    expect(wrapper.find('.is-animating')).toBeTruthy()
    wrapper.vm.translateItem(2)
    expect(wrapper.vm.active).toBeTruthy()
    expect(wrapper.get('.el-carousel__item').attributes('style')).toContain(
      'translateY(0px)'
    )
  })
})
