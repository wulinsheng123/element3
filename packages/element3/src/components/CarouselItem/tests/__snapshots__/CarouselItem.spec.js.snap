// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel.vue snapshot 1`] = `
<div
  class="el-carousel__item"
  style="transform: translateX(0px);"
>
  <div
    class="el-carousel__mask"
  />
  
  
</div>
`;
